### PR TITLE
⬆️🔒️ Maintenance: starlette-stack updates due to vulnerabilities

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- 🔨 Fixes requirements/tools ``make reqs-all`` ( issue reported by @sanderegg )
   - Adds ``make reqs-services`` recipe 
- 🔒️ Fixes https://github.com/advisories/GHSA-qj8w-rv5x-2v9h
- ⬆️ upgrades all ``fastapi`` to the same version
   -  NOTE that it  replaced `requests` by `httpx` !
   - ♻️ `request.url_for -> URL` and not `srt`. Moved common dependency to ``serverilib``
   - ~♻️ `director-v2` forward dependency is now illegal. Created a wrapper~
      - REVERTED upgrade in directorv2. SEE https://github.com/ITISFoundation/osparc-simcore/issues/4332 
   - ♻️ `dynamic-sidecar`  openapi.json updated

###  Highlights on updated libraries (only updated libraries are included)

- #packages before: 4
- #packages after : 2

| #   | name      | before                                         | after     | upgrade | count | packages                                                                                                                                                     |
| --- | --------- | ---------------------------------------------- | --------- | ------- | ----- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| 1   | fastapi   | 0.90.1, 0.85.1, 0.85.0, 0.89.1, 0.95.2, 0.85.2 | 0.96.0    | *minor* | 9     | agent⬆️</br>api-server⬆️</br>autoscaling⬆️</br>catalog⬆️</br>datcore-adapter⬆️</br>director-v2⬆️</br>dynamic-sidecar⬆️</br>invitations⬆️</br>resource-usage-tracker⬆️ |
| 2   | requests  | 2.31.0                                         | 🗑️ removed |         | 3     | api-server⬆️</br>catalog⬆️</br>director-v2⬆️                                                                                                                    |
| 3   | starlette | 0.20.4, 0.23.1, 0.22.0                         | 0.27.0    | *minor* | 8     | agent⬆️</br>api-server⬆️</br>autoscaling⬆️</br>catalog⬆️</br>datcore-adapter⬆️</br>director-v2⬆️</br>dynamic-sidecar⬆️</br>invitations⬆️                             |
| 4   | urllib3   | 1.26.9                                         | 🗑️ removed |         | 2     | api-server⬆️</br>catalog⬆️                                                                                                                                     |

*Legend*:
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

### Repo-wide overview of libraries
- #reqs files parsed: 76

| #   | name                     | versions-base                                     | versions-test                                     | versions-tool                            |
| --- | ------------------------ | ------------------------------------------------- | ------------------------------------------------- | ---------------------------------------- |
| 1   | aio-pika                 | 9.1.2                                             | 9.1.2                                             |                                          |
| 2   | aioboto3                 | 9.6.0, 10.4.0                                     | 9.6.0, 11.2.0                                     |                                          |
| 3   | aiobotocore              | 2.3.0, 2.4.2, 2.5.0                               | 2.3.0, 2.5.0                                      |                                          |
| 4   | aiocache                 | 0.11.1, 0.12.1                                    | 0.12.1                                            |                                          |
| 5   | aiodebug                 | 2.3.0                                             | 2.3.0                                             |                                          |
| 6   | aiodocker                | 0.19.1, 0.21.0                                    | 0.21.0                                            |                                          |
| 7   | aiofiles                 | 0.8.0, 22.1.0, 23.1.0                             | 23.1.0                                            |                                          |
| 8   | aiohttp                  | 3.8.3, 3.8.4                                      | 3.8.3, 3.8.4                                      |                                          |
| 9   | aiohttp-jinja2           | 1.5                                               |                                                   |                                          |
| 10  | aiohttp-security         | 0.4.0                                             |                                                   |                                          |
| 11  | aiohttp-session          | 2.11.0                                            |                                                   |                                          |
| 12  | aiohttp-swagger          | 1.0.16                                            |                                                   |                                          |
| 13  | aioitertools             | 0.10.0, 0.11.0                                    | 0.11.0                                            |                                          |
| 14  | aiopg                    | 1.4.0                                             | 1.4.0                                             |                                          |
| 15  | aioprocessing            | 2.0.1                                             |                                                   |                                          |
| 16  | aioredis                 | 2.0.1                                             |                                                   |                                          |
| 17  | aioresponses             |                                                   | 0.7.4                                             |                                          |
| 18  | aiormq                   | 6.7.6                                             | 6.7.6                                             |                                          |
| 19  | aiosignal                | 1.2.0, 1.3.1                                      | 1.2.0, 1.3.1                                      |                                          |
| 20  | aiosmtplib               | 1.1.6                                             |                                                   |                                          |
| 21  | aiozipkin                | 1.1.1                                             |                                                   |                                          |
| 22  | alembic                  | 1.8.1, 1.11.1                                     | 1.8.1, 1.11.1                                     |                                          |
| 23  | anyio                    | 3.6.2                                             | 3.6.2, 3.7.0                                      |                                          |
| 24  | arrow                    | 1.2.3                                             | 1.2.3                                             |                                          |
| 25  | asgi-lifespan            |                                                   | 2.1.0                                             |                                          |
| 26  | asgiref                  | 3.5.2                                             |                                                   |                                          |
| 27  | astroid                  |                                                   |                                                   | 2.15.5                                   |
| 28  | async-asgi-testclient    |                                                   | 1.4.11                                            |                                          |
| 29  | async-timeout            | 4.0.2                                             | 4.0.2                                             |                                          |
| 30  | asyncpg                  | 0.27.0                                            |                                                   |                                          |
| 31  | attrs                    | 21.4.0, 23.1.0                                    | 21.4.0, 23.1.0                                    |                                          |
| 32  | aws-sam-translator       |                                                   | 1.55.0, 1.68.0                                    |                                          |
| 33  | aws-xray-sdk             |                                                   | 2.12.0                                            |                                          |
| 34  | bcrypt                   | 3.2.0                                             |                                                   |                                          |
| 35  | bidict                   | 0.22.0                                            |                                                   |                                          |
| 36  | black                    |                                                   |                                                   | 23.3.0                                   |
| 37  | blinker                  |                                                   | 1.6.2                                             |                                          |
| 38  | blosc                    | 1.11.1                                            |                                                   |                                          |
| 39  | bokeh                    | 2.4.3                                             | 2.4.3                                             |                                          |
| 40  | boto3                    | 1.21.21, 1.24.59, 1.24.96                         | 1.21.21, 1.24.59, 1.26.76, 1.26.147               |                                          |
| 41  | boto3-stubs              |                                                   | 1.26.147                                          |                                          |
| 42  | botocore                 | 1.24.21, 1.27.59, 1.27.96, 1.29.76                | 1.24.21, 1.27.59, 1.29.76, 1.29.147               |                                          |
| 43  | botocore-stubs           | 1.27.17, 1.29.78                                  | 1.29.147                                          |                                          |
| 44  | build                    |                                                   |                                                   | 0.10.0                                   |
| 45  | bump2version             |                                                   |                                                   | 1.0.1                                    |
| 46  | certifi                  | 2022.12.7, 2023.5.7                               | 2022.12.7, 2023.5.7                               |                                          |
| 47  | cffi                     | 1.15.0, 1.15.1                                    | 1.15.0, 1.15.1                                    |                                          |
| 48  | cfgv                     |                                                   |                                                   | 3.3.1                                    |
| 49  | cfn-lint                 |                                                   | 0.72.0, 0.72.6, 0.77.6                            |                                          |
| 50  | change-case              |                                                   |                                                   | 0.5.2                                    |
| 51  | charset-normalizer       | 2.0.12, 2.1.1, 3.0.1, 3.1.0                       | 2.0.12, 2.1.1, 3.0.1, 3.1.0                       |                                          |
| 52  | click                    | 8.1.3                                             | 8.1.3                                             | 8.1.3                                    |
| 53  | cloudpickle              | 2.2.1                                             | 2.2.1                                             |                                          |
| 54  | colorama                 | 0.4.6                                             |                                                   |                                          |
| 55  | colorlog                 | 6.7.0                                             | 6.7.0                                             |                                          |
| 56  | commonmark               | 0.9.1                                             |                                                   |                                          |
| 57  | contourpy                | 1.0.7                                             |                                                   |                                          |
| 58  | coverage                 |                                                   | 7.2.7                                             |                                          |
| 59  | cryptography             | 39.0.1, 40.0.2                                    | 39.0.1, 41.0.1                                    |                                          |
| 60  | cycler                   | 0.11.0                                            |                                                   |                                          |
| 61  | dask                     | 2023.3.2, 2023.5.1                                | 2023.3.2                                          |                                          |
| 62  | dask-gateway             | 2023.1.1                                          | 2023.1.1                                          |                                          |
| 63  | dask-gateway-server      | 2023.1.1                                          | 2023.1.1                                          |                                          |
| 64  | dateparser               | 1.1.8                                             |                                                   |                                          |
| 65  | debugpy                  |                                                   | 1.6.7                                             |                                          |
| 66  | decorator                | 4.4.2                                             |                                                   |                                          |
| 67  | deepdiff                 |                                                   | 6.3.0                                             |                                          |
| 68  | dill                     |                                                   |                                                   | 0.3.6                                    |
| 69  | distlib                  |                                                   |                                                   | 0.3.6                                    |
| 70  | distributed              | 2023.3.2, 2023.5.1                                | 2023.3.2                                          |                                          |
| 71  | distro                   | 1.5.0                                             |                                                   |                                          |
| 72  | dnspython                | 2.1.0, 2.2.1, 2.3.0                               | 2.3.0                                             |                                          |
| 73  | docker                   | 6.0.0, 6.1.3                                      | 6.1.3                                             |                                          |
| 74  | docker-compose           | 1.29.1                                            |                                                   |                                          |
| 75  | dockerpty                | 0.4.1                                             |                                                   |                                          |
| 76  | docopt                   | 0.6.2                                             |                                                   |                                          |
| 77  | ecdsa                    |                                                   | 0.18.0                                            |                                          |
| 78  | email-validator          | 1.2.1, 1.3.0, 1.3.1, 2.0.0.post2                  | 2.0.0.post2                                       |                                          |
| 79  | et-xmlfile               | 1.1.0                                             |                                                   |                                          |
| 80  | exceptiongroup           | 1.1.1                                             | 1.1.1                                             |                                          |
| 81  | execnet                  |                                                   | 1.9.0                                             |                                          |
| 82  | expiringdict             | 1.2.1                                             |                                                   |                                          |
| 83  | faker                    |                                                   | 18.10.1                                           |                                          |
| 84  | fakeredis                |                                                   | 2.14.0                                            |                                          |
| 85  | fastapi                  | 0.96.0                                            |                                                   |                                          |
| 86  | fastapi-pagination       | 0.10.0                                            |                                                   |                                          |
| 87  | filelock                 |                                                   |                                                   | 3.12.0                                   |
| 88  | flaky                    |                                                   | 3.7.0                                             |                                          |
| 89  | flask                    |                                                   | 2.1.3, 2.2.5, 2.3.2                               |                                          |
| 90  | flask-cors               |                                                   | 3.0.10                                            |                                          |
| 91  | fonttools                | 4.39.4                                            |                                                   |                                          |
| 92  | frozenlist               | 1.3.0, 1.3.1, 1.3.3                               | 1.3.0, 1.3.1, 1.3.3                               |                                          |
| 93  | fsspec                   | 2023.5.0                                          | 2023.5.0                                          |                                          |
| 94  | graphql-core             |                                                   | 3.2.3                                             |                                          |
| 95  | greenlet                 | 2.0.2                                             | 2.0.2                                             |                                          |
| 96  | gunicorn                 | 20.1.0                                            |                                                   |                                          |
| 97  | h11                      | 0.12.0, 0.14.0                                    | 0.12.0, 0.14.0                                    |                                          |
| 98  | h2                       | 4.1.0                                             |                                                   |                                          |
| 99  | hpack                    | 4.0.0                                             |                                                   |                                          |
| 100 | httmock                  | 1.4.0                                             |                                                   |                                          |
| 101 | httpcore                 | 0.15.0, 0.16.3, 0.17.1                            | 0.15.0, 0.16.3, 0.17.1, 0.17.2                    |                                          |
| 102 | httptools                | 0.2.0, 0.5.0                                      |                                                   |                                          |
| 103 | httpx                    | 0.24.0, 0.24.1                                    | 0.24.0, 0.24.1                                    |                                          |
| 104 | hyperframe               | 6.0.1                                             |                                                   |                                          |
| 105 | hypothesis               |                                                   | 6.76.0                                            |                                          |
| 106 | icdiff                   |                                                   | 2.0.6                                             |                                          |
| 107 | identify                 |                                                   |                                                   | 2.5.24                                   |
| 108 | idna                     | 2.10, 3.3, 3.4                                    | 2.10, 3.3, 3.4                                    |                                          |
| 109 | importlib-metadata       | 6.6.0                                             | 6.6.0                                             |                                          |
| 110 | iniconfig                | 2.0.0                                             | 2.0.0                                             |                                          |
| 111 | inotify                  |                                                   |                                                   | 0.2.10                                   |
| 112 | isodate                  | 0.6.1                                             |                                                   |                                          |
| 113 | isort                    |                                                   |                                                   | 5.12.0                                   |
| 114 | itsdangerous             | 1.1.0, 2.1.2                                      | 2.1.2                                             |                                          |
| 115 | jinja-app-loader         | 1.0.2                                             |                                                   |                                          |
| 116 | jinja2                   | 3.1.2                                             | 3.1.2                                             | 3.1.2                                    |
| 117 | jmespath                 | 1.0.0, 1.0.1                                      | 1.0.0, 1.0.1                                      |                                          |
| 118 | jschema-to-python        |                                                   | 1.2.3                                             |                                          |
| 119 | json2html                | 1.3.0                                             |                                                   |                                          |
| 120 | jsondiff                 | 2.0.0                                             | 2.0.0                                             |                                          |
| 121 | jsonpatch                |                                                   | 1.32                                              |                                          |
| 122 | jsonpickle               |                                                   | 3.0.1                                             |                                          |
| 123 | jsonpointer              |                                                   | 2.3                                               |                                          |
| 124 | jsonschema               | 3.2.0, 4.17.3                                     | 3.2.0, 4.17.3                                     |                                          |
| 125 | junit-xml                |                                                   | 1.9                                               |                                          |
| 126 | kiwisolver               | 1.4.4                                             |                                                   |                                          |
| 127 | lazy-object-proxy        | 1.7.1                                             |                                                   | 1.7.1, 1.9.0                             |
| 128 | locket                   | 1.0.0                                             | 1.0.0                                             |                                          |
| 129 | lupa                     |                                                   | 1.14.1                                            |                                          |
| 130 | lz4                      | 4.3.2                                             | 4.3.2                                             |                                          |
| 131 | mako                     | 1.2.2, 1.2.4                                      | 1.2.2, 1.2.4                                      |                                          |
| 132 | markdown-it-py           | 2.2.0                                             |                                                   |                                          |
| 133 | markupsafe               | 2.1.1, 2.1.2, 2.1.3                               | 2.1.1, 2.1.2, 2.1.3                               | 2.1.1                                    |
| 134 | matplotlib               | 3.7.1                                             |                                                   |                                          |
| 135 | mccabe                   |                                                   |                                                   | 0.7.0                                    |
| 136 | mdurl                    | 0.1.2                                             |                                                   |                                          |
| 137 | minio                    |                                                   | 7.0.4                                             |                                          |
| 138 | moto                     |                                                   | 4.0.1, 4.1.11                                     |                                          |
| 139 | mpmath                   |                                                   | 1.3.0                                             |                                          |
| 140 | msgpack                  | 1.0.3, 1.0.5                                      | 1.0.5                                             |                                          |
| 141 | multidict                | 6.0.2, 6.0.3, 6.0.4                               | 6.0.2, 6.0.4                                      |                                          |
| 142 | mypy                     |                                                   | 1.3.0                                             |                                          |
| 143 | mypy-extensions          |                                                   | 1.0.0                                             | 1.0.0                                    |
| 144 | networkx                 | 2.5.1                                             | 2.8.8, 3.1                                        |                                          |
| 145 | nodeenv                  |                                                   |                                                   | 1.8.0                                    |
| 146 | nose                     |                                                   |                                                   | 1.3.7                                    |
| 147 | numpy                    | 1.24.3                                            | 1.24.3                                            |                                          |
| 148 | openapi-core             | 0.12.0                                            |                                                   |                                          |
| 149 | openapi-schema-validator | 0.2.3                                             | 0.2.3                                             |                                          |
| 150 | openapi-spec-validator   | 0.4.0                                             | 0.4.0                                             |                                          |
| 151 | openpyxl                 | 3.0.9                                             |                                                   |                                          |
| 152 | ordered-set              | 4.1.0                                             | 4.1.0                                             |                                          |
| 153 | orjson                   | 3.7.2                                             |                                                   |                                          |
| 154 | packaging                | 23.0, 23.1                                        | 23.0, 23.1                                        | 23.0, 23.1                               |
| 155 | pamqp                    | 3.2.1                                             | 3.2.1                                             |                                          |
| 156 | pandas                   | 2.0.1                                             | 2.0.2                                             |                                          |
| 157 | paramiko                 | 2.11.0                                            |                                                   |                                          |
| 158 | partd                    | 1.4.0                                             | 1.4.0                                             |                                          |
| 159 | passlib                  | 1.7.4                                             |                                                   |                                          |
| 160 | pathspec                 |                                                   |                                                   | 0.11.1                                   |
| 161 | pbr                      |                                                   | 5.11.1                                            |                                          |
| 162 | pillow                   | 9.5.0                                             | 9.5.0                                             |                                          |
| 163 | pint                     | 0.19.2, 0.22                                      | 0.22                                              |                                          |
| 164 | pip-tools                |                                                   |                                                   | 6.13.0                                   |
| 165 | platformdirs             |                                                   |                                                   | 3.5.1                                    |
| 166 | pluggy                   | 1.0.0                                             | 1.0.0                                             |                                          |
| 167 | pprintpp                 |                                                   | 0.4.0                                             |                                          |
| 168 | pre-commit               |                                                   |                                                   | 3.3.2                                    |
| 169 | prometheus-api-client    | 0.5.3                                             |                                                   |                                          |
| 170 | prometheus-client        | 0.14.1                                            |                                                   |                                          |
| 171 | psutil                   | 5.9.1, 5.9.5                                      | 5.9.5                                             |                                          |
| 172 | psycopg2-binary          | 2.9.6                                             | 2.9.6                                             |                                          |
| 173 | ptvsd                    |                                                   | 4.3.2                                             |                                          |
| 174 | py-cpuinfo               |                                                   | 9.0.0                                             |                                          |
| 175 | py-partiql-parser        |                                                   | 0.3.3                                             |                                          |
| 176 | pyasn1                   |                                                   | 0.5.0                                             |                                          |
| 177 | pycparser                | 2.20, 2.21                                        | 2.21                                              |                                          |
| 178 | pydantic                 | 1.9.0, 1.10.2, 1.10.7, 1.10.8                     | 1.10.2, 1.10.8                                    |                                          |
| 179 | pyftpdlib                |                                                   | 1.5.7                                             |                                          |
| 180 | pygments                 | 2.13.0, 2.14.0, 2.15.1                            |                                                   |                                          |
| 181 | pyinstrument             | 3.4.2, 4.1.1, 4.3.0, 4.4.0, 4.5.0                 | 4.5.0                                             |                                          |
| 182 | pyinstrument-cext        | 0.2.4                                             |                                                   |                                          |
| 183 | pyjwt                    | 2.4.0                                             |                                                   |                                          |
| 184 | pylint                   |                                                   |                                                   | 2.17.4                                   |
| 185 | pynacl                   | 1.4.0                                             |                                                   |                                          |
| 186 | pyopenssl                |                                                   | 23.2.0                                            |                                          |
| 187 | pyparsing                | 3.0.9                                             | 3.0.9                                             |                                          |
| 188 | pyproject-hooks          |                                                   |                                                   | 1.0.0                                    |
| 189 | pyrsistent               | 0.18.1, 0.19.2, 0.19.3                            | 0.18.1, 0.19.2, 0.19.3                            |                                          |
| 190 | pytest                   | 7.3.1                                             | 7.3.1                                             |                                          |
| 191 | pytest-aiohttp           |                                                   | 1.0.4                                             |                                          |
| 192 | pytest-asyncio           |                                                   | 0.21.0                                            |                                          |
| 193 | pytest-benchmark         |                                                   | 4.0.0                                             |                                          |
| 194 | pytest-cov               |                                                   | 4.1.0                                             |                                          |
| 195 | pytest-docker            |                                                   | 1.0.1                                             |                                          |
| 196 | pytest-icdiff            |                                                   | 0.6                                               |                                          |
| 197 | pytest-instafail         |                                                   | 0.5.0                                             |                                          |
| 198 | pytest-lazy-fixture      |                                                   | 0.6.3                                             |                                          |
| 199 | pytest-localftpserver    |                                                   | 1.1.4                                             |                                          |
| 200 | pytest-mock              |                                                   | 3.10.0                                            |                                          |
| 201 | pytest-runner            |                                                   | 6.0.0                                             |                                          |
| 202 | pytest-sugar             |                                                   | 0.9.7                                             |                                          |
| 203 | pytest-xdist             |                                                   | 3.3.1                                             |                                          |
| 204 | python-dateutil          | 2.8.2                                             | 2.8.2                                             |                                          |
| 205 | python-dotenv            | 0.20.0, 0.21.0, 1.0.0                             | 0.21.0, 1.0.0                                     |                                          |
| 206 | python-engineio          | 4.3.4                                             |                                                   |                                          |
| 207 | python-jose              |                                                   | 3.3.0                                             |                                          |
| 208 | python-magic             | 0.4.25                                            |                                                   |                                          |
| 209 | python-multipart         | 0.0.5                                             |                                                   |                                          |
| 210 | python-socketio          | 5.8.0                                             |                                                   |                                          |
| 211 | pytz                     | 2022.1, 2023.3                                    | 2023.3                                            |                                          |
| 212 | pyyaml                   | 5.4.1, 6.0                                        | 5.4.1, 6.0                                        | 5.4.1, 6.0                               |
| 213 | redis                    | 4.5.4, 4.5.5                                      | 4.5.4, 4.5.5                                      |                                          |
| 214 | regex                    | 2023.5.5                                          | 2023.6.3                                          |                                          |
| 215 | requests                 | 2.30.0, 2.31.0                                    | 2.30.0, 2.31.0                                    |                                          |
| 216 | requests-mock            |                                                   | 1.10.0                                            |                                          |
| 217 | responses                |                                                   | 0.23.1                                            |                                          |
| 218 | respx                    |                                                   | 0.20.1                                            |                                          |
| 219 | rich                     | 12.5.1, 12.6.0, 13.3.5, 13.4.1                    |                                                   |                                          |
| 220 | rsa                      |                                                   | 4.9                                               |                                          |
| 221 | s3fs                     | 2023.5.0                                          |                                                   |                                          |
| 222 | s3transfer               | 0.5.2, 0.6.0                                      | 0.5.2, 0.6.0, 0.6.1                               |                                          |
| 223 | sarif-om                 |                                                   | 1.0.4                                             |                                          |
| 224 | semantic-version         | 2.9.0                                             |                                                   |                                          |
| 225 | setproctitle             | 1.2.3                                             |                                                   |                                          |
| 226 | shellingham              | 1.5.0.post1                                       |                                                   |                                          |
| 227 | six                      | 1.15.0, 1.16.0                                    | 1.15.0, 1.16.0                                    |                                          |
| 228 | sniffio                  | 1.2.0, 1.3.0                                      | 1.2.0, 1.3.0                                      |                                          |
| 229 | sortedcontainers         | 2.4.0                                             | 2.4.0                                             |                                          |
| 230 | sqlalchemy               | 1.4.47, 1.4.48                                    | 1.4.47, 1.4.48                                    |                                          |
| 231 | sqlalchemy2-stubs        |                                                   | 0.0.2a34                                          |                                          |
| 232 | sshpubkeys               |                                                   | 3.3.1                                             |                                          |
| 233 | starlette                | 0.27.0                                            |                                                   |                                          |
| 234 | strict-rfc3339           | 0.7                                               |                                                   |                                          |
| 235 | sympy                    |                                                   | 1.12                                              |                                          |
| 236 | tblib                    | 1.7.0                                             | 1.7.0                                             |                                          |
| 237 | tenacity                 | 8.0.1, 8.1.0, 8.2.1, 8.2.2                        | 8.0.1, 8.2.2                                      |                                          |
| 238 | termcolor                |                                                   | 2.3.0                                             |                                          |
| 239 | texttable                | 1.6.3                                             |                                                   |                                          |
| 240 | tomli                    | 2.0.1                                             | 2.0.1                                             | 2.0.1                                    |
| 241 | tomlkit                  |                                                   |                                                   | 0.11.8                                   |
| 242 | toolz                    | 0.12.0                                            | 0.12.0                                            |                                          |
| 243 | tornado                  | 6.3.2                                             | 6.3.2                                             |                                          |
| 244 | tqdm                     | 4.64.0, 4.64.1, 4.65.0                            | 4.65.0                                            |                                          |
| 245 | traitlets                | 5.9.0                                             | 5.9.0                                             |                                          |
| 246 | twilio                   | 7.12.0                                            |                                                   |                                          |
| 247 | typer                    | 0.4.1, 0.6.1, 0.7.0, 0.9.0                        | 0.9.0                                             | 0.9.0                                    |
| 248 | types-aiobotocore        | 2.3.3, 2.4.2.post1                                |                                                   |                                          |
| 249 | types-aiobotocore-ec2    | 2.4.2                                             |                                                   |                                          |
| 250 | types-aiobotocore-s3     | 2.3.3                                             |                                                   |                                          |
| 251 | types-aiofiles           |                                                   | 23.1.0.3                                          |                                          |
| 252 | types-awscrt             | 0.16.10                                           | 0.16.19                                           |                                          |
| 253 | types-boto3              |                                                   | 1.0.2                                             |                                          |
| 254 | types-pkg-resources      |                                                   | 0.1.3                                             |                                          |
| 255 | types-pyyaml             |                                                   | 6.0.12.10                                         |                                          |
| 256 | types-s3transfer         |                                                   | 0.6.1                                             |                                          |
| 257 | typing-extensions        | 4.3.0, 4.4.0, 4.5.0, 4.6.0, 4.6.1, 4.6.3          | 4.3.0, 4.4.0, 4.5.0, 4.6.0, 4.6.1, 4.6.3          | 4.3.0, 4.4.0, 4.5.0, 4.6.0, 4.6.1, 4.6.3 |
| 258 | tzdata                   | 2023.3                                            | 2023.3                                            |                                          |
| 259 | tzlocal                  | 5.0.1                                             |                                                   |                                          |
| 260 | ujson                    | 5.5.0                                             |                                                   |                                          |
| 261 | urllib3                  | 1.26.9, 1.26.11, 1.26.12, 1.26.14, 1.26.16, 2.0.2 | 1.26.9, 1.26.11, 1.26.12, 1.26.14, 1.26.16, 2.0.2 |                                          |
| 262 | uvicorn                  | 0.15.0, 0.17.0, 0.19.0, 0.20.0, 0.22.0            |                                                   |                                          |
| 263 | uvloop                   | 0.16.0, 0.17.0                                    |                                                   |                                          |
| 264 | virtualenv               |                                                   |                                                   | 20.23.0                                  |
| 265 | watchdog                 | 2.1.5                                             |                                                   | 3.0.0                                    |
| 266 | watchfiles               | 0.18.0, 0.18.1, 0.19.0                            |                                                   |                                          |
| 267 | watchgod                 | 0.8.2                                             |                                                   |                                          |
| 268 | websocket-client         | 0.59.0, 1.5.2                                     | 0.59.0, 1.5.2                                     |                                          |
| 269 | websockets               | 10.1, 10.2, 10.3, 10.4, 11.0.3                    | 11.0.3                                            |                                          |
| 270 | werkzeug                 | 2.1.2, 2.2.2                                      | 2.1.2, 2.2.2, 2.3.4                               |                                          |
| 271 | wheel                    |                                                   |                                                   | 0.40.0                                   |
| 272 | wrapt                    | 1.14.1, 1.15.0                                    | 1.14.1, 1.15.0                                    | 1.14.1, 1.15.0                           |
| 273 | xmltodict                |                                                   | 0.13.0                                            |                                          |
| 274 | yarl                     | 1.5.1, 1.9.2                                      | 1.5.1, 1.9.2                                      |                                          |
| 275 | zict                     | 3.0.0                                             | 3.0.0                                             |                                          |
| 276 | zipp                     | 3.15.0                                            | 3.15.0                                            |                                          |

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
quite the upgrade

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/api-server/src/simcore_service_api_server/api/dependencies/application.py [hunk 1]
```diff
@@ -33,3 +20,12 @@ async def get_product_name(app: FastAPI = Depends(get_app)) -> str:
...
     return default_product_name
+
+
+assert get_reverse_url_mapper  # nosec
+assert get_app  # nosec
+
+__all__: tuple[str, ...] = (
+    "get_reverse_url_mapper",
+    "get_app",
+)
```
Comment (GitHK) [inline services/api-server/src/simcore_service_api_server/api/dependencies/application.py:31]: Q: why can't those dependencies be imported where they are used from servicelib?

File: services/datcore-adapter/src/simcore_service_datcore_adapter/api/dependencies/application.py [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
-def get_reverse_url_mapper(request: Request) -> Callable[..., str]:
-    def reverse_url_mapper(name: str, **path_params: Any) -> str:
-        # NOTE: the cast appears to be needed by mypy
-        return cast(str, request.url_for(name, **path_params))
-
-    return reverse_url_mapper
+__all__: tuple[str, ...] = (
+    "get_reverse_url_mapper",
+    "get_app",
+)
```
Comment (GitHK) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/api/dependencies/application.py:9]: `get_app` was not previously present here, I would avoid adding it.
Also same question as above

File: services/invitations/src/simcore_service_invitations/api/_dependencies.py [hunk 3]
```diff
@@ -57,3 +48,13 @@ def _is_valid(current: str, expected: str) -> bool:
...
+
+
+assert get_reverse_url_mapper  # nosec
+assert get_app  # nosec
+
+
+__all__: tuple[str, ...] = (
+    "get_reverse_url_mapper",
+    "get_app",
+)
```
Comment (GitHK) [inline services/invitations/src/simcore_service_invitations/api/_dependencies.py:60]: Same two questions as above here as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/api/_dependencies.py [hunk 4]
```diff
@@ -1,20 +1,14 @@
...
-        url: URL = request.url_for(name, **path_params)
-        return f"{url}"
+assert get_reverse_url_mapper  # nosec
+assert get_app  # nosec
 
-    return _reverse_url_mapper
+__all__: tuple[str, ...] = (
+    "get_reverse_url_mapper",
+    "get_app",
+)
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/api/_dependencies.py:14]: and again here

**Review by GitHK — APPROVED** [review]
Nice.
Only one thing bothers me, how decencies are reimported when using `get_app` and `get_reverse_url_mapper`. I don't see any benefit in importing them inside the decencies module. If you look at the path, when importing from servicelib, you still see they are a fastapi dependency, which I think is very clear.

File: services/api-server/src/simcore_service_api_server/api/dependencies/application.py (hunk 1 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/dependencies/application.py:31]: I use it as "cluster for common dependencies" in a service.  Note that all services have the same layout/skeleton.
let me follow up on that in coming PRs after having discussed a bit with all BE to find a consensus, ok?

**Pull request merged**